### PR TITLE
Added upload tests for Zextras-Drive

### PR DESCRIPTION
**Pull request by zmashish**
Have Added Drive related classes in Pages and Test folder and have also modified few of the classes.

### CODE REVIEW

File: src/java/com/zimbra/qa/selenium/projects/ajax/pages/PageMain.java [hunk 1]
```diff
@@ -42,6 +42,7 @@
 		public static final String zCalendarApp = "id=zb__App__Calendar_title";
 		public static final String zTasksApp = "id=zb__App__Tasks_title";
 		public static final String zBriefcaseApp = "css=td[id=zb__App__Briefcase_title]";
+		public static final String zDriveApp = "css=td[id=zb__App__ZIMBRA_DRIVE_title]";
```
Comment (jiteshsojitra) [inline src/java/com/zimbra/qa/selenium/projects/ajax/pages/PageMain.java]: Can you please move at bottom?

File: src/java/com/zimbra/qa/selenium/projects/ajax/tests/zextras/UploadFile.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
+/*
```
Comment (jiteshsojitra) [inline src/java/com/zimbra/qa/selenium/projects/ajax/tests/zextras/UploadFile.java]: Can you create folder "drive" under zextras and move testcase file over there?

File: src/java/com/zimbra/qa/selenium/projects/ajax/tests/zextras/UploadFile.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+public class UploadFile extends AjaxCore{
+
+	public UploadFile() throws HarnessException {
+		logger.info("New " + UploadFile.class.getCanonicalName());
+		super.startingPage = app.zPageDrive;
+		super.startingAccountPreferences.put("zimbraPrefShowSelectionCheckbox","TRUE");
+	}
+
+	@Test (description = "Upload file through GUI - verify through GUI",
```
Comment (jiteshsojitra) [inline src/java/com/zimbra/qa/selenium/projects/ajax/tests/zextras/UploadFile.java]: Please remove "test" test group.

**Comment by jiteshsojitra** [comment]
Code looks good, thanks Ashish. Can you attach test result as well?

**Comment by zmashish** [comment]
image

File: .classpath [hunk 4]
```diff
@@ -6,5 +6,7 @@
 	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER"/>
 	<classpathentry kind="con" path="org.apache.ivyde.eclipse.cpcontainer.IVYDE_CONTAINER/?project=zm-selenium&amp;ivyXmlPath=ivy.xml&amp;confs=*"/>
 	<classpathentry kind="lib" path="build/dist/lib/ZimbraSelenium.jar"/>
+	<classpathentry kind="lib" path="C:/Users/ashishk/git/zm-mailbox/native/build/zm-native-8.7.6.1510740065.jar"/>
```
Comment (jiteshsojitra) [inline .classpath]: Please remove .classpath change.

File: src/java/com/zimbra/qa/selenium/projects/ajax/tests/zextras/drive/UploadFile.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+public class UploadFile extends AjaxCore{
+
+	public UploadFile() throws HarnessException {
+		logger.info("New " + UploadFile.class.getCanonicalName());
+		super.startingPage = app.zPageDrive;
+		super.startingAccountPreferences.put("zimbraPrefShowSelectionCheckbox","TRUE");
+	}
+
+	@Test (description = "Upload file through UI and verify it",
+			groups = { "sanity", "L0", "upload", "drive","test" })
```
Comment (jiteshsojitra) [inline src/java/com/zimbra/qa/selenium/projects/ajax/tests/zextras/drive/UploadFile.java]: Again, please remove "test" group.

File: src/java/com/zimbra/qa/selenium/projects/ajax/core/AjaxCore.java [hunk 6]
```diff
@@ -603,6 +606,8 @@ public void zUploadFile(String filePath, String attachmentType) throws HarnessEx
 						we = webDriver.findElement(By.name("__calAttUpload__"));
 					} else if (isBriefcaseApp == true) {
 						we = webDriver.findElement(By.name("uploadFile"));
+					} else if (isZextrasDrive == true) {
```
Comment (jiteshsojitra) [inline src/java/com/zimbra/qa/selenium/projects/ajax/core/AjaxCore.java]: It should be isZimbraDrive instead of isZextrasDrive everywhere.

**Review by jiteshsojitra — APPROVED** [review]

**Pull request merged**